classlib/src/main/java/org/teavm/classlib/java/lang/TString.java

```diff
@@ -122,6 +127,7 @@ public TString(int[] codePoints, int offset, int count) {
     }
 
     private void initWithBytes(byte[] bytes, int offset, int length, TCharset charset) {
+        checkSIOOB(offset, length, bytes.length);
```

Comment:
Is this check really necessary? Won't further code throw some exception when indexes are invalid?

Reply:
It will throw (and throws), but exception is incorrect.

Reply:
In 99.99999% of situations there's no difference at all. I can hardly imagine some code, except for synthetic tests for JVM spec conformance, that really relies on SIOOB/AIOOB exception type and makes such difference.

Reply:
As an alternative you can also wrap AIIOB (or whatever exception is thrown) and re-throw SIOOB. Exceptions, that aren't thrown, have zero cost. There's still some impact on JavaScript size though.

Reply:
So you are suggesting to use the same `Objects.checkFromIndexSize`?

Reply:
No, I suggest not to perform any check at all. The line below uses ByteBuffer.wrap, which performs the very same check: https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/nio/ByteBuffer.html#wrap(byte%5B%5D,int,int)

Reply:
Added missing check to ByteBuffer instead and removed everywhere where bytes are used.